File: challenges/3/3.py
def is_prime(n):
    # check if n is a prime, return bool
    for i in range(2, int(n**(.5))+1):
        if n % i == 0:
            return False
    return True


def prime_list(n):
    # create list of primes up to value n
    prime_list = []
    for i in range(n):
        if is_prime(i):
            prime_list.append(i)
    return prime_list


def prime_factor(n, pl):
    # create list of prime numbers
    prime_nums = prime_list(pl)

    # return largest prime factor by returning
    # 1st value in a reversed list
    for i in reversed(prime_nums):
        if n % i == 0:
            return i


print(prime_factor(600851475143, 10000))

File: challenges/3/3.js
function is_prime(n) {
    // check if n is a prime, return bool
    for (let i = 2; i < Math.ceil(n ** 0.5) + 1; i++) {
        if (n % i == 0) {
            return false;
        }
    }
    return true;
}

function prime_list(n) {
    // create an array of primes up to value n
    primeList = [2];
    for (let i = 2; i < (n+1); i++) {
        if (is_prime(i)) {
            primeList.push(i);
        }
    }
    return primeList;
}

function prime_factor(n, pl) {
    /*
        n:  number to find largest prime factor for
        pl: create a list of all prime numbers that fall
            within the range of 1 and this value
    */

    // create list of primes, upto & including value of pl
    prime_nums = prime_list(pl)

    // return largest prime factor by returning
    // 1st factor in a reversed list of primes
    for (let i = prime_nums.length - 1; i >= 0; i--) {
        if (n % prime_nums[i] == 0) {
            return prime_nums[i]
        }
    }
}

console.log(prime_factor(600851475143, 10000))
